services/server/server.go

```diff
@@ -215,7 +217,10 @@ func New(ctx context.Context, config *srvconfig.Config) (*Server, error) {
 		)
 		initContext.Address = config.GRPC.Address
 		initContext.TTRPCAddress = config.TTRPC.Address
-		initContext.RegisterReadiness = s.RegisterReadiness
+		initContext.RegisterReadiness = func() func() {
+			atomic.StoreInt32(&mustSucceed, 1)
```

Comment:
we could, I don't think sync really needs to be used here anyway since init happens synchronously and `RegisterReadiness` shouldn't be called in another goroutine. I was wondering about the error messaging though or whether we should have a debug log in that case indicating the plugin is added as a required plugin. As it is now, it would be confusing if it failed for that reason when the configuration is empty.